control-plane/config-entries/controllersv2/meshconfig_controller_test.go

```diff
@@ -145,6 +145,11 @@ func TestMeshConfigController_createsMeshConfig(t *testing.T) {
 			resourceClient, err := consul.NewResourceServiceClient(testClient.Watcher)
 			require.NoError(t, err)
 
+			require.Eventually(t, func() bool {
+				_, _, err := testClient.APIClient.Partitions().Read(context.Background(), constants.DefaultConsulPartition, nil)
+				return err == nil
+			}, 5*time.Second, 500*time.Millisecond)
```

Comment:
~~ Could helper-ify this since it's needed a bunch of places

Reply:
That's a fair comment, but I didn't want to find a home for this if eventually I can get it back in the SDK.